gen_initramfs.sh

```diff
@@ -100,7 +100,13 @@ append_base_layout() {
 	date -u '+%Y%m%d-%H%M%S' > ${TEMP}/initramfs-base-temp/etc/build_date
 	echo "Genkernel $GK_V" > ${TEMP}/initramfs-base-temp/etc/build_id
 
-	printf "$(hostid | sed 's/\([0-9A-F]\{2\}\)/\\x\1/gI')" > ${TEMP}/initramfs-base-temp/etc/hostid
+	little_endian="$(echo I | tr -d [:space:] | od -to2 | head -n1 | awk '{print $2}' | cut -c6)"
+	if test "$little_endian" = 1; then
+	    hostid="$(hostid |sed 's/\(..\)\(..\)\(..\)\(..\)/\4\3\2\1/')"
```

Comment:
space before the `|`, and use the -E option to sed so you don't have to escape every `(` and `)`.